Description: pre-requisite for upstream-D134738.diff to apply cleanly
From: https://hg.mozilla.org/mozilla-central/rev/3607c4937e84
Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1747862

# HG changeset patch
# User Landry Breuil <landry@openbsd.org>
# Date 1640882964 -3600
# Node ID 3607c4937e84ccf8b3e39f7b40e664cca2448c22
# Parent  436f3c0ae63b68202c949bf48ef5769153f85812
Bug 1747862 - regenerate webrtc moz.build files after bug 1744644. r=mjf

diff --git a/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn/moz.build b/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn/moz.build
--- a/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn/moz.build
+++ b/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn/moz.build
@@ -79,16 +79,27 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
     DEFINES["_GNU_SOURCE"] = True
     DEFINES["_LARGEFILE64_SOURCE"] = True
     DEFINES["_LARGEFILE_SOURCE"] = True
     DEFINES["__STDC_CONSTANT_MACROS"] = True
     DEFINES["__STDC_FORMAT_MACROS"] = True
 
+if CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["USE_GLIB"] = "1"
+    DEFINES["WEBRTC_BSD"] = True
+    DEFINES["WEBRTC_POSIX"] = True
+    DEFINES["_FILE_OFFSET_BITS"] = "64"
+    DEFINES["_LARGEFILE64_SOURCE"] = True
+    DEFINES["_LARGEFILE_SOURCE"] = True
+    DEFINES["__STDC_CONSTANT_MACROS"] = True
+    DEFINES["__STDC_FORMAT_MACROS"] = True
+
 if CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["CERT_CHAIN_PARA_HAS_EXTRA_FIELDS"] = True
     DEFINES["NOMINMAX"] = True
     DEFINES["NTDDI_VERSION"] = "0x0A000000"
     DEFINES["PSAPI_VERSION"] = "2"
     DEFINES["UNICODE"] = True
     DEFINES["USE_AURA"] = "1"
@@ -141,28 +152,36 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Darwin":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_DEBUG"] = True
 
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Darwin":
 
     DEFINES["_DEBUG"] = True
 
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["_DEBUG"] = True
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_DEBUG"] = True
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
         "-msse2"
diff --git a/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn/moz.build b/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn/moz.build
--- a/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn/moz.build
+++ b/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn/moz.build
@@ -79,16 +79,27 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
     DEFINES["_GNU_SOURCE"] = True
     DEFINES["_LARGEFILE64_SOURCE"] = True
     DEFINES["_LARGEFILE_SOURCE"] = True
     DEFINES["__STDC_CONSTANT_MACROS"] = True
     DEFINES["__STDC_FORMAT_MACROS"] = True
 
+if CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["USE_GLIB"] = "1"
+    DEFINES["WEBRTC_BSD"] = True
+    DEFINES["WEBRTC_POSIX"] = True
+    DEFINES["_FILE_OFFSET_BITS"] = "64"
+    DEFINES["_LARGEFILE64_SOURCE"] = True
+    DEFINES["_LARGEFILE_SOURCE"] = True
+    DEFINES["__STDC_CONSTANT_MACROS"] = True
+    DEFINES["__STDC_FORMAT_MACROS"] = True
+
 if CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["CERT_CHAIN_PARA_HAS_EXTRA_FIELDS"] = True
     DEFINES["NOMINMAX"] = True
     DEFINES["NTDDI_VERSION"] = "0x0A000000"
     DEFINES["PSAPI_VERSION"] = "2"
     DEFINES["UNICODE"] = True
     DEFINES["USE_AURA"] = "1"
@@ -141,28 +152,36 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Darwin":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_DEBUG"] = True
 
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Darwin":
 
     DEFINES["_DEBUG"] = True
 
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["_DEBUG"] = True
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_DEBUG"] = True
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
         "-msse2"
diff --git a/third_party/libwebrtc/moz.build b/third_party/libwebrtc/moz.build
--- a/third_party/libwebrtc/moz.build
+++ b/third_party/libwebrtc/moz.build
@@ -245,16 +245,18 @@ DIRS += [
     "/third_party/libwebrtc/modules/third_party/fft/fft_gn",
     "/third_party/libwebrtc/modules/third_party/g711/g711_3p_gn",
     "/third_party/libwebrtc/modules/third_party/g722/g722_3p_gn",
     "/third_party/libwebrtc/modules/utility/utility_gn",
     "/third_party/libwebrtc/modules/video_capture/video_capture_internal_impl_gn",
     "/third_party/libwebrtc/modules/video_capture/video_capture_module_gn",
     "/third_party/libwebrtc/modules/video_coding/chain_diff_calculator_gn",
     "/third_party/libwebrtc/modules/video_coding/codec_globals_headers_gn",
+    "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn",
+    "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn",
     "/third_party/libwebrtc/modules/video_coding/deprecated/nack_module_gn",
     "/third_party/libwebrtc/modules/video_coding/encoded_frame_gn",
     "/third_party/libwebrtc/modules/video_coding/frame_dependencies_calculator_gn",
     "/third_party/libwebrtc/modules/video_coding/nack_module_gn",
     "/third_party/libwebrtc/modules/video_coding/video_codec_interface_gn",
     "/third_party/libwebrtc/modules/video_coding/video_coding_gn",
     "/third_party/libwebrtc/modules/video_coding/video_coding_utility_gn",
     "/third_party/libwebrtc/modules/video_coding/webrtc_h264_gn",
@@ -361,53 +363,42 @@ DIRS += [
     "/third_party/libwebrtc/third_party/rnnoise/rnn_vad_gn",
     "/third_party/libwebrtc/video/adaptation/video_adaptation_gn",
     "/third_party/libwebrtc/video/frame_dumping_decoder_gn",
     "/third_party/libwebrtc/video/video_gn",
     "/third_party/libwebrtc/video/video_stream_encoder_impl_gn",
     "/third_party/libwebrtc/webrtc_gn"
 ]
 
-if CONFIG["OS_TARGET"] == "Android":
-
-    DIRS += [
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn",
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn"
-    ]
-
 if CONFIG["OS_TARGET"] == "Darwin":
 
     DIRS += [
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_decoder_isac_float_gn",
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_encoder_isac_float_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_c_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_gn",
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn",
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_objc_gn",
         "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn",
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn",
         "/third_party/libwebrtc/rtc_base/system/cocoa_threading_gn",
         "/third_party/libwebrtc/sdk/base_objc_gn",
         "/third_party/libwebrtc/sdk/helpers_objc_gn"
     ]
 
 if CONFIG["OS_TARGET"] == "Linux":
 
     DIRS += [
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_decoder_isac_float_gn",
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_encoder_isac_float_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_c_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_gn",
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn",
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
-        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn",
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn"
+        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn"
     ]
 
 if CONFIG["OS_TARGET"] == "OpenBSD":
 
     DIRS += [
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_decoder_isac_float_gn",
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_encoder_isac_float_gn",
         "/third_party/libwebrtc/common_audio/common_audio_avx2_gn",
@@ -426,19 +417,17 @@ if CONFIG["OS_TARGET"] == "WINNT":
 
     DIRS += [
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_decoder_isac_float_gn",
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_encoder_isac_float_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_c_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_gn",
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn",
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
-        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_decoder_gn",
-        "/third_party/libwebrtc/modules/video_coding/codecs/av1/libaom_av1_encoder_gn"
+        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn"
     ]
 
 if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
 
     DIRS += [
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_decoder_isac_float_gn",
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_encoder_isac_float_gn",
         "/third_party/libwebrtc/common_audio/common_audio_neon_c_gn",

